### PR TITLE
Cloudbuilds for each push to main branch

### PR DESCRIPTION
**Pull request by rjbez17**
This is the initial cloudbuild file that will be triggered on every merge to the main branch. I'll follow up later (haven't started this work yet) adding manifests and krew plugins to the artifact store as well. 

Additionally, I removed the currently not working github action as we are taking a different approach.

### CODE REVIEW

**Comment by adrianludwin** [comment]
How does Github automatically trigger this? Or is this just a preparatory step?

**Comment by rjbez17** [comment]
I gave the k8s-staging-multitenacy cloudbuilder authorization to read our GitHub repo. So it's watching for pushes.

**Comment by adrianludwin** [comment]
Hey Ryan, can you please add some comments in the new yaml file explaining what you described to me - e.g. that we have triggers set up in the `k8s-staging-multitenacy` project, and how you did that? Maybe also update the last section of `README.md` as well, i.e. where we talk about the Prow config. It's probably useful to have a little description about how we granted authorization (or even why authorization is needed, given that it's a public repo). Thanks!

File: docs/automated-builds.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# Automated Builds
+
+HNC uses [Google Cloud Build](https://cloud.google.com/build) for building some artifacts.
+Currently only docker builds are automated for every push to the main branch.
+
+## Trigger Setup
```
Comment (adrianludwin) [inline docs/automated-builds.md:6]: Can you add the name of the GCP project as well as who owns it, so we can get permission to add new people in the future if needed?

Thanks!

**Review by adrianludwin — COMMENTED** [review]
One last comment then lgtm

File: docs/automated-builds.md (hunk 1 above)
Comment (rjbez17) [reply docs/automated-builds.md:6]: done

**Comment by adrianludwin** [comment]
Oh, cool. Maybe document this in the new file? E.g. "This is monitored by
Cloud Build in the k8s-staging-multitenancy project"?

**Pull request merged**